alicloud/resource_alicloud_oss_bucket.go

```diff
@@ -969,6 +1028,34 @@ func resourceAlicloudOssBucketDelete(d *schema.ResourceData, meta interface{}) e
 			return nil, ossClient.DeleteBucket(d.Id())
 		})
 		if err != nil {
+			if err.(oss.ServiceError).Code == "BucketNotEmpty" {
+				if d.Get("force_destroy").(bool) {
+					log.Printf("[DEBUG] oss Bucket attempting to forceDestroy %+v", err)
+					client.WithOssClient(func(ossClient *oss.Client) (interface{}, error) {
+						bucket, _ := ossClient.Bucket(d.Get("bucket").(string))
+						lor, err := bucket.ListObjectVersions()
+						if err == nil {
+							objectsToDelete := make([]oss.DeleteObject, 0)
+							for _, object := range lor.ObjectDeleteMarkers {
+								objectsToDelete = append(objectsToDelete, oss.DeleteObject{
+									Key:       object.Key,
+									VersionId: object.VersionId,
+								})
+							}
+
+							for _, object := range lor.ObjectVersions {
+								objectsToDelete = append(objectsToDelete, oss.DeleteObject{
+									Key:       object.Key,
+									VersionId: object.VersionId,
+								})
+							}
+
+							bucket.DeleteObjectVersions(objectsToDelete)
```

Comment:
Does the method `DeleteObjectVersions ` return an error?

Reply:
add err  and check it.